apps/desktop/src/lib/components/IntegrateUpstreamModal.svelte

```diff
@@ -0,0 +1,239 @@
+<script lang="ts">
+	import { BaseBranchService } from '$lib/baseBranch/baseBranchService';
+	import CommitCard from '$lib/commit/CommitCard.svelte';
+	import { getGitHost } from '$lib/gitHost/interface/gitHost';
+	import ScrollableContainer from '$lib/scroll/ScrollableContainer.svelte';
+	import Select from '$lib/select/Select.svelte';
+	import SelectItem from '$lib/select/SelectItem.svelte';
+	import { getContext } from '$lib/utils/context';
+	import {
+		getResolutionApproach,
+		sortStatusInfo,
+		UpstreamIntegrationService,
+		type BranchStatusesWithBranches,
+		type BranchStatusInfo,
+		type Resolution
+	} from '$lib/vbranches/upstreamIntegrationService';
+	import Button from '@gitbutler/ui/Button.svelte';
+	import Modal from '@gitbutler/ui/Modal.svelte';
+	import { tick } from 'svelte';
+	import { SvelteMap } from 'svelte/reactivity';
+	import type { Readable } from 'svelte/store';
+
+	type OperationState = 'inert' | 'loading' | 'completed';
+
+	interface Props {
+		onClose?: () => void;
+	}
+
+	const { onClose }: Props = $props();
+
+	const gitHost = getGitHost();
+	const upstreamIntegrationService = getContext(UpstreamIntegrationService);
+	let branchStatuses = $state<Readable<BranchStatusesWithBranches | undefined>>();
+	const baseBranchService = getContext(BaseBranchService);
+	const base = baseBranchService.base;
+
+	let modal = $state<Modal>();
+	let integratingUpstream = $state<OperationState>('inert');
+	let results = $state(new SvelteMap<string, Resolution>());
+	let statuses = $state<BranchStatusInfo[]>([]);
+	let expanded = $state<boolean>(false);
+
+	$effect(() => {
+		if ($branchStatuses?.type !== 'updatesRequired') {
+			statuses = [];
+			return;
+		}
+
+		const statusesTmp = [...$branchStatuses.subject];
+		statusesTmp.sort(sortStatusInfo);
+
+		// Side effect, refresh results
+		results = new SvelteMap(
+			statusesTmp.map((status) => {
+				const defaultApproach = getResolutionApproach(status);
+
+				return [
+					status.branch.id,
+					{
+						branchId: status.branch.id,
+						branchTree: status.branch.tree,
+						approach: defaultApproach
+					}
+				];
+			})
+		);
+
+		statuses = statusesTmp;
+	});
+
+	async function integrate() {
+		integratingUpstream = 'loading';
+		await tick();
+		await upstreamIntegrationService.integrateUpstream(Array.from(results.values()));
+		await baseBranchService.refresh();
+		integratingUpstream = 'completed';
+
+		modal?.close();
+	}
+
+	export async function integrateOrShow() {
+		integratingUpstream = 'inert';
+		expanded = false;
+		branchStatuses = upstreamIntegrationService.upstreamStatuses();
+		await tick();
+		if (branchStatuses) {
+			modal?.show();
+			return;
+		}
+		await integrate();
+	}
```

Comment:
This triggers the integration of the upstream changes.
If there are no upstream statuses returned, i.e there are no foreseeable conflicts, the changes are integrated.
If there are foreseeable conflicts, display the modal.

What do we think about this approach?
I like the fact that this treats the `IntegrateUpstreamModal` component as a class with methods as is intended in Svelte, but I'm not sure if this is something we like as a pattern for this specific case. 
Let me know :)